### ISSUE TITLE
Code folding is broken with go_highlight_types 

### ISSUE THREAD

**Issue by jeromer** [fatih/vim-go#1282]
### Behavior

Code folding is broken when `go_highlight_types` is enabled.

### Steps to reproduce:

Take the following code:

```
package main

import "fmt"

type someStruct struct {
	A string
	B string
}

func main() {
	test()
}

func test() {
	s := someStruct{A: "a", B: "b"}
	fmt.Println(s)
}
```

And the following vim configuration

```
set nocompatible

filetype off
set rtp+=~/.vim/bundle/vundle
call vundle#rc()

Bundle 'fatih/vim-go'

set directory=/tmp/

syntax enable

set fdm=syntax

let g:go_fmt_command = "goimports"
let g:go_autodetect_gopath = 1
let g:go_list_type = "quickfix"
let g:go_def_mapping_enabled = 0

"let g:go_highlight_types = 1
```

Everything looks OK.

![a](https://cloud.githubusercontent.com/assets/93039/26240408/3ae35f04-3c81-11e7-8f14-2302c5ea9d8e.png)

Now uncomment `let g:go_highlight_types = 1` quit vim and reopen the .go file. The `test` function is incorrectly folded.

![b](https://cloud.githubusercontent.com/assets/93039/26240448/58caeb18-3c81-11e7-99f5-1e359e6c43e5.png)

### Configuration

Add here your current configuration and additional information that might be
useful, such as:

* `vimrc` you used to reproduce : see above
* vim version: 7.4.712
* vim-go version: v1.12
* go version: go1.8 linux/amd64

**Comment by Carpetsmoker** [comment]
I think I fixed it in #1338 @jeromer. Could you test this? I don't use either feature in my daily usage.

**Comment by jeromer** [comment]
Confirmed, your commit fixed that bug.

Thanks :)
